packages/edit-site/src/components/style-book/button-examples.tsx

```diff
@@ -0,0 +1,81 @@
+/**
+ * WordPress dependencies
+ */
+import { createBlock } from '@wordpress/blocks';
+import { __experimentalGrid as Grid } from '@wordpress/components';
+import { View } from '@wordpress/primitives';
+import {
+	BlockList,
+	privateApis as blockEditorPrivateApis,
+} from '@wordpress/block-editor';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { unlock } from '../../lock-unlock';
+
+const { ExperimentalBlockEditorProvider, useGlobalStyle } = unlock(
+	blockEditorPrivateApis
+);
+
+const BUTTON_STATES = [
+	{
+		key: 'default',
+		title: __( 'Button' ),
+	},
+	{
+		key: ':active',
+		title: __( 'Button (Active)' ),
+	},
+	{
+		key: ':any-link',
+		title: __( 'Button (Any Link)' ),
+	},
+	{
+		key: ':focus',
+		title: __( 'Button (Focus)' ),
+	},
+	{
+		key: ':hover',
+		title: __( 'Button (Hover)' ),
+	},
+	{
+		key: ':link',
+		title: __( 'Button (Link)' ),
+	},
+	{
+		key: ':visited',
+		title: __( 'Button (Visited)' ),
+	},
+];
+
+function ButtonExamples() {
+	const [ elementsButton ] = useGlobalStyle( 'elements.button' );
+	const blocks = BUTTON_STATES.map( ( { key } ) => {
+		const styles =
+			( key !== 'default' ? elementsButton[ key ] : elementsButton ) ||
```

Comment:
What happens when we have both a "default" and a "state", are we merging the state styles with the "default" styles (I just want to confirm that this is covered because it's not clear by reading this code)